### PR TITLE
spdm: add single measurement fetching

### PR DESCRIPTION
**Pull request by twilfredo**

### CODE REVIEW

File: src/libspdm/spdm.rs [hunk 1]
```diff
@@ -784,6 +840,16 @@ pub unsafe fn get_measurement(
...
     }
 
+    // Measurement Manifest
+    if measurement_index == SPDM_MEASUREMENT_BLOCK_MEASUREMENT_INDEX_MEASUREMENT_MANIFEST {
+        // Let's attempt to decode this (assuming CBOR encoding),
+        // a decoding failure likely implies that this was not CBOR encoded/serialised properly.
+
+        let measurement_manifest = &measurement_record
+            [LIBSPDM_MANIFEST_RAW_BITSTREAM_OFFSET..measurement_record_length as usize];
+        _ = process_cbor_measurement_manifest(measurement_manifest).unwrap();
```
Comment (alistair23) [inline src/libspdm/spdm.rs:850]: Should this be an `unwrap()`?

Comment (twilfredo) [reply src/libspdm/spdm.rs]: fixed!

File: src/libspdm/spdm.rs [hunk 2]
```diff
@@ -649,54 +721,9 @@ pub unsafe fn get_measurements(context: *mut c_void, slot_id: u8) -> Result<(),
...
-                        "\n{}",
-                        String::from_utf8_lossy(&decoded_bytes).blue().bold()
-                    );
-                    info!(
-                        "---Decoded measurement manifest {} end---",
-                        "diagnostic".blue()
-                    );
-                }
-                Err(_) => warn!("Failed to decode manifest into diagnostic format"),
+            if let Err(_) = process_cbor_measurement_manifest(measurement_manifest) {
```
Comment (alistair23) [inline src/libspdm/spdm.rs:724]:
```
if process_cbor_measurement_manifest(measurement_manifest).is_error() {
```

Comment (twilfredo) [reply src/libspdm/spdm.rs:724]: what's the difference?

Comment (twilfredo) [reply src/libspdm/spdm.rs]: changed to suggestion

Comment (alistair23) [reply src/libspdm/spdm.rs]: If we aren't using the error value then we don't need to assign it. I think it's something Clippy will complain about if we ran it

**Review by alistair23 — APPROVED** [review]

**Pull request merged**